### PR TITLE
Add DKB Broker support

### LINKED ISSUE #5: Support für DKB
Issue für DKB Support

### PR DESCRIPTION
**Pull request by d-a-n**
- addresses #5
- improves the way brokers are loaded
- adds Jest for unit tests
- enhances project setup by introducing eslint, prettier and enables pre-commit hooks

Tresor one PDF Import Demo 2020-04-23 02-31-09

tresor-import — -zsh — 254×74 2020-04-23 02-29-05

### CODE REVIEW

**Comment by codeofsumit** [comment]
Hi Dan, just merged trade republic support, could you resolve the merge conflicts?
Will test this afterwards. If anyone has DKB files to test for me, feel free to send some to me sumit@tresor.one

**Comment by bauerj** [comment]
Looks great!

One minor point: It would be great if you would test that no other handlers want to parse your test files using getBroker in index.js.

**Comment by d-a-n** [comment]
@codeofsumit rebased my PR

> no other handlers want to parse your test files using getBroker in index.js

@bauerj could you elaborate what you mean?

**Comment by bauerj** [comment]
Of course. You wrote a method that returns the applicable brokers for a given text input. It would be helpful to test that the test files for one broker only match that specific broker.

For example, the OnVista Bank has a text like "eine Tochtergesellschaft der comdirect Bank" in their PDFs. The OnVista tests should check that no other broker handler matches their test input. And I think that would be true for any broker. For example:

```js
expect(getBroker(sample)).toEqual(["DKB"])
```

**Comment by d-a-n** [comment]
@bauerj I've added 3 more tests. Is this what you had in mind?

https://github.com/codeofsumit/tresor-import/pull/13/files#diff-3149133ef3685a9f95dcfbdcea13d7ef

**Comment by bauerj** [comment]
Yes, exactly! But I think it would be good to also test against the actual PDF content from the JSON files.

**Comment by d-a-n** [comment]
At the moment we have DKB samples only. Happy to iterate over all as soon as we have samples for other brokers.

**Comment by bauerj** [comment]
Sure, that makes sense.

**Comment by codeofsumit** [comment]
let's keep the branches and PRs clean and have this one for everything DKB. Tests for other brokers into new branches please 🙏

**Comment by codeofsumit** [comment]
alright as this PR adds some groundwork, it HAS to be merged before any other branch.
Please resolve the merge conflicts and assign me when it's ready for review 🙏 

Good work so far 👍  - Looking forward to it!

**Comment by d-a-n** [comment]
@codeofsumit I've resolved the merge conflict. I have no permissions to assign it to you.

**Comment by codeofsumit** [comment]
Ah ok - so the PR is ready from your end? If so, I will review & test tomorrow. I have (very) limited DKB files so if anyone can send me some to sumit@tresor.one - it's appreciated 👍

**Comment by codeofsumit** [comment]
Wie siehts hier aus @d-a-n ?

**Comment by d-a-n** [comment]
@codeofsumit Thanks for providing more samples. I added them to the test suite and adjusted the DKB importer accordingly.

File: src/brokers/dkb.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  textArr.some(t => t.includes('Dividendengutschrift'));
+
+export const canParseData = textArr =>
+  textArr.some(t => t.includes('BIC BYLADEM1001')) &&
+  (isBuy(textArr) || isSell(textArr) || isDividend(textArr));
+
+export const parseData = textArr => {
+  let type, date, isin, company, shares, price, amount, fee;
+
+  if (isBuy(textArr)) {
```
Comment (codeofsumit) [inline src/brokers/dkb.js:73]: Would be awesome to support the `tax` field too. See `traderepublic.js` for reference

**Comment by codeofsumit** [comment]
@d-a-n werd asap mergen wenn tax support mit dabei is 👍

**Comment by codeofsumit** [comment]
@d-a-n merging this now as this PR is required for some other work.
Adding `tax` support as a new issue.
Good work, man! 👏👏

**Comment by codeofsumit** [comment]
Hatte hier bei meiner Review was übersehen.
`throw` hab ich jetzt in https://github.com/codeofsumit/tresor-import/commit/93efb6b2edbeadc08a1e74a56826692ff6230f5e wieder rausgenommen. Grund ist, dass die JS execution nicht gestoppt werden soll wenn kein Broker gefunden wird sondern entsprechend values returned werden sollen, so dass ich dem User anzeigen kann was passiert.

**Pull request merged**